### ISSUE TITLE
-Wcompare-distinct-pointer-types in kernel/sched/cpufreq_schedutil.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#420]
```
kernel/sched/cpufreq_schedutil.c:346:26: warning: comparison of distinct pointer types ('typeof (sg_cpu->iowait_boost << 1) *' (aka 'unsigned int *') and 'typeof ((1L << 10)) *' (aka 'long *')) [-Wcompare-distinct-pointer-types]
                sg_cpu->iowait_boost = min(sg_cpu->iowait_boost << 1, SCHED_CAPACITY_SCALE);
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
include/linux/kernel.h:860:19: note: expanded from macro 'min'
#define min(x, y)       __careful_cmp(x, y, <)
                        ^~~~~~~~~~~~~~~~~~~~~~
include/linux/kernel.h:851:24: note: expanded from macro '__careful_cmp'
        __builtin_choose_expr(__safe_cmp(x, y), \
                              ^~~~~~~~~~~~~~~~
include/linux/kernel.h:841:4: note: expanded from macro '__safe_cmp'
                (__typecheck(x, y) && __no_side_effects(x, y))
                 ^~~~~~~~~~~~~~~~~
include/linux/kernel.h:827:29: note: expanded from macro '__typecheck'
                (!!(sizeof((typeof(x) *)1 == (typeof(y) *)1)))
                           ~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~
1 warning generated.
```

**Comment by nathanchance** [comment]
The maintainers are aware of this one, I am not sure of the proper fix so I'll let them sort it out and keep an eye on this: https://lore.kernel.org/lkml/CAHk-=wi_krg9B2B0i7+mV4rRCdowiNw3bOC9rssaZTVFQvKMxQ@mail.gmail.com/

**Comment by nathanchance** [comment]
This has been fixed in an updated version of this patch, should be fixed in today's -next version: https://git.kernel.org/tip/a23314e9d88d89d49e69db08f60b7caa470f04e1

**Comment by nathanchance** [comment]
Fixed in `next-20190320`: https://git.kernel.org/next/linux-next/c/a23314e9d88d89d49e69db08f60b7caa470f04e1
